### PR TITLE
[Constraint] Make name of constraints more explicit

### PR DESCRIPTION
**Pull request by bakpaul**
This PR is a first step for the issue #3952. 

I modified the names of the constraints following the rule that we agreed upon. Still need to change spring force-fields that can behave as penalties. 

The changes where straight, not more though than adding the type into the name was done. If any name doesn't fit your standard, feel free to suggest any changes. 

This PR also add a componentRenamed mechanism that creates an alias automatically for renamed objects.

TODO:

- [x]  Edit Sofa/framework/Helper/src/sofa/helper/ComponentChange.cpp
  - [x]  Add a API in ComponentChange.cpp for Renamed component (avoiding to add an alias manually in the class)
- [x] Move the C++ alias in the old header
- [x] Change name of projective constraints strating by `ProjectOn`

### CODE REVIEW

**Comment by bakpaul** [comment]
[ci-build][with-all-tests]

**Comment by bakpaul** [comment]
I am just not sure about the alias compatibility. It is for scene launching, but nothing guaranty that those aliases will one day be removed. Maybe the factory lacks a method `addDeprecationAlias` (or I am not aware of this) ?

**Review by alxbilger — COMMENTED** [review]
You need to edit Sofa/framework/Helper/src/sofa/helper/ComponentChange.cpp

**Comment by bakpaul** [comment]
[ci-build][with-all-tests]

File: Sofa/Component/Constraint/Lagrangian/Model/src/sofa/component/constraint/lagrangian/model/SlidingLagrangianConstraint.cpp [hunk 1]
```diff
@@ -32,10 +32,10 @@ namespace sofa::component::constraint::lagrangian::model
 using namespace sofa::defaulttype;
 using namespace sofa::helper;
 
-int SlidingConstraintClass = core::RegisterObject("TODO-SlidingConstraint")
-        .add< SlidingConstraint<Vec3Types> >(true);
+int SlidingLagrangianConstraintClass = core::RegisterObject("TODO-SlidingLagrangianConstraint")
```
Comment (alxbilger) [inline Sofa/Component/Constraint/Lagrangian/Model/src/sofa/component/constraint/lagrangian/model/SlidingLagrangianConstraint.cpp:35]: I am not really happy with this description...

File: applications/plugins/SofaCarving/examples/CarvingTool.scn [hunk 2]
```diff
@@ -5,7 +5,7 @@
         <RequiredPlugin name="Sofa.Component.Collision.Detection.Intersection"/> <!-- Needed to use components [MinProximityIntersection] -->  
         <RequiredPlugin name="Sofa.Component.Collision.Geometry"/> <!-- Needed to use components [PointCollisionModel, TriangleCollisionModel] -->  
         <RequiredPlugin name="Sofa.Component.Collision.Response.Contact"/> <!-- Needed to use components [CollisionResponse] -->  
-        <RequiredPlugin name="Sofa.Component.Constraint.Projective"/> <!-- Needed to use components [FixedConstraint, LinearMovementConstraint] -->  
+        <RequiredPlugin name="Sofa.Component.Constraint.Projective"/> <!-- Needed to use components [FixedConstraint, LinearMovementProjectiveConstraint] -->
```
Comment (alxbilger) [inline applications/plugins/SofaCarving/examples/CarvingTool.scn]:
```suggestion
        <RequiredPlugin name="Sofa.Component.Constraint.Projective"/> <!-- Needed to use components [LinearMovementProjectiveConstraint] -->
```

There is also an occurence of `FixedConstraint` in this file. Could you check again in the examples of all the code base that you renamed the components?

**Review by alxbilger — CHANGES_REQUESTED** [review]
Almost good to go. Keep in mind that the documentation website will need to be updated accordingly.

File: Sofa/Component/Constraint/Lagrangian/Model/src/sofa/component/constraint/lagrangian/model/SlidingLagrangianConstraint.cpp (hunk 1 above)
Comment (bakpaul) [reply Sofa/Component/Constraint/Lagrangian/Model/src/sofa/component/constraint/lagrangian/model/SlidingLagrangianConstraint.cpp:35]: Yes me neither, but in fact this is not the only component and it is not really in the scope of this pr...

**Comment by bakpaul** [comment]
[ci-build][with-all-tests]

**Comment by bakpaul** [comment]
Finally I needed the changed introduced in the (now closed) [PR #4324](https://github.com/sofa-framework/sofa/pull/4324) to finalize the automatic alias creation mechanism. Now CI should not be crying anymore.

**Comment by bakpaul** [comment]
[ci-build][with-all-tests]

**Pull request merged**